rebar.config.script

```diff
@@ -15,6 +15,18 @@
 ConfigureEnv = filename:join(filename:dirname(SCRIPT), "config.erl"),
 os:putenv("COUCHDB_CONFIG", ConfigureEnv).
 
+OTP_VSN = erlang:system_info(compat_rel),
+
+Features = lists:filter(fun({_K, Flag}) -> Flag end, [
+    {'HAS_CALLBACKS', OTP_VSN > 14}
+]),
+
+Properties = Features ++ [
+    {'OTP_VSN', OTP_VSN},
+    {'OTP_RELEASE', erlang:system_info(otp_release)}
+],
```

Comment:
The way I see it, we need to define a set of constants in order to make version-dependent macros later on. crypto:hash/2 was added in R15B02, for example;

```
-ifdef(VERSION_15B02_OR_HIGHER).
-define(HASH(Alg, Data),
 	crypto:hash(Alg, Data)).
-else.
-define(HASH(Alg, Data),
 	case Alg of
            sha -> crypto:sha(Data);
            md5 -> crypto:md5(Data)
	end).
-endif.

hash(Alg, Data) ->
    ?HASH(Alg, Data).
```